### PR TITLE
CowSdk options params to enable 0x and Paraswap APIs

### PR DESCRIPTION
**Pull request by W3stside**
CowSdk has options parameter to enable `ZeroXApi` and `ParaswapApi`

1. Adjusts tests
2. Small update to `ParaswapApi` constructor to allow passing options
3. Rename instances of `Matcha` to be `ZeroX`

### CODE REVIEW

File: src/CowSdk.ts [hunk 1]
```diff
@@ -6,29 +6,37 @@ import { SupportedChainId as ChainId, SupportedChainId } from './constants/chain
...
 }
 
 export class CowSdk<T extends ChainId> {
   context: Context
   cowApi: CowApi
   metadataApi: MetadataApi
   cowSubgraphApi: CowSubgraphApi
-  zeroXApi: ZeroXApi
-  paraswapApi: ParaswapApi
+  zeroXApi?: ZeroXApi
```
Comment (W3stside) [inline src/CowSdk.ts:25]: PR #60 fixes this entire types debacle here with the optional APIs as well as the tests where the `?` operator is used with the apis. e.g `cowSdk.paraswapApi?.getQuote(...)`

File: src/api/0x/0x-quote.spec.ts [hunk 2]
```diff
@@ -96,7 +98,7 @@ describe('Get Quote', () => {
       }
     )
     // WHEN - we fetch a quote
-    const price = await cowSdk.zeroXApi.getQuote(query)
+    const price = await cowSdk.zeroXApi!.getQuote(query)
```
Comment (W3stside) [inline src/api/0x/0x-quote.spec.ts:101]: `!` assertion removed in #60 from proper typing

File: src/api/0x/index.ts [hunk 3]
```diff
@@ -25,29 +25,29 @@ const EXCLUDED_SOURCES: ERC20BridgeSource[] = []
...
-    { affiliateAddressMap, excludedSources }: MatchaOptions = {
-      affiliateAddressMap: GP_SETTLEMENT_CONTRACT_ADDRESS,
-      excludedSources: EXCLUDED_SOURCES,
-    }
-  ) {
+  constructor(chainId: SupportedChainId, options: Partial<ZeroXOptions> | undefined) {
     super({ context: new Context(chainId, {}), name: '0x', apiVersion: API_VERSION, getApiUrl: get0xUrls })
 
+    // get defaults if necessary
+    const affiliateAddressMap = options?.affiliateAddressMap || GP_SETTLEMENT_CONTRACT_ADDRESS
```
Comment (W3stside) [inline src/api/0x/index.ts:35]: allow options to set these in constructor

**Review by alfetopito — APPROVED** [review]

File: src/CowSdk.ts [hunk 4]
```diff
@@ -6,29 +6,37 @@ import { SupportedChainId as ChainId, SupportedChainId } from './constants/chain
...
+import { ZeroXOptions } from './api/0x/types'
 import ParaswapApi from './api/paraswap'
+import { ParaswapOptions } from './api/paraswap/types'
+import { WithEnabled } from './types'
 
 type Options = {
   loglevel?: LogLevelDesc
-  matchaOptions?: MatchaOptions
+  zeroXOptions?: Partial<ZeroXOptions & WithEnabled>
+  paraswapOptions?: Partial<ParaswapOptions> & WithEnabled
```
Comment (ramirotw) [inline src/CowSdk.ts:17]: here the intersection of `WithEnabled` is done inside the Partial in `zeroXOptions` and outside in `paraswapOptions`. Both work the same as `enabled` is optional by default but it makes you think.

File: src/CowSdk.ts [hunk 5]
```diff
@@ -6,29 +6,37 @@ import { SupportedChainId as ChainId, SupportedChainId } from './constants/chain
...
   metadataApi: MetadataApi
   cowSubgraphApi: CowSubgraphApi
-  zeroXApi: ZeroXApi
-  paraswapApi: ParaswapApi
+  zeroXApi?: ZeroXApi
+  paraswapApi?: ParaswapApi
 
   constructor(chainId: T = SupportedChainId.MAINNET as T, cowContext: CowContext = {}, options: Options = {}) {
+    const zeroXEnabled = !!options?.zeroXOptions?.enabled
+    const paraswapEnabled = !!options?.paraswapOptions?.enabled
```
Comment (ramirotw) [inline src/CowSdk.ts:30]:
```suggestion
    const paraswapEnabled = options?.paraswapOptions?.enabled ?? false
```

File: src/CowSdk.ts [hunk 6]
```diff
@@ -6,29 +6,37 @@ import { SupportedChainId as ChainId, SupportedChainId } from './constants/chain
...
   cowApi: CowApi
   metadataApi: MetadataApi
   cowSubgraphApi: CowSubgraphApi
-  zeroXApi: ZeroXApi
-  paraswapApi: ParaswapApi
+  zeroXApi?: ZeroXApi
+  paraswapApi?: ParaswapApi
 
   constructor(chainId: T = SupportedChainId.MAINNET as T, cowContext: CowContext = {}, options: Options = {}) {
+    const zeroXEnabled = !!options?.zeroXOptions?.enabled
```
Comment (ramirotw) [inline src/CowSdk.ts:29]:
```suggestion
    const zeroXEnabled = options?.zeroXOptions?.enabled ?? false
```

**Review by ramirotw — APPROVED** [review]

File: src/CowSdk.ts (hunk 4 above)
Comment (W3stside) [reply src/CowSdk.ts:17]: hahah shit. ok good point let me fix this. good 👁️

Comment (W3stside) [reply src/CowSdk.ts:17]: done in #60

**Pull request merged**